src/lib/deploy.js

```diff
@@ -243,8 +247,33 @@ async function deploy({ alias, yes }) {
   const verificationKey = await step(
     'Generate verification key (takes 10-30 sec)',
     async () => {
-      let { verificationKey } = await zkApp.compile(zkAppAddress);
-      return verificationKey;
+      let cache = fs.readJsonSync(`${DIR}/build/cache.json`);
+      // compute a hash of the contract's circuit to determine if 'zkapp.compile' should re-run
+      let currentDigest = await zkApp.digest(zkAppAddress);
+
+      // initialize cache if 'zk deploy' is run the first time on the contract
+      if (!cache[contractName]) {
+        cache[contractName] = { digest: '', verificationKey: '' };
+        fs.writeJsonSync(`${DIR}/build/cache.json`, cache, { spaces: 2 });
+      }
+
+      if (cache[contractName]['digest'] === currentDigest) {
+        let cacheKey = fs.readJSONSync(`${DIR}/build/cache.json`)[contractName]
+          .verificationKey;
+        console.log('Using the cached verification key');
```

Comment:
Does this text need to contain leading spaces (2?) to maintain consistent indentation in the CLI's output?

Reply:
Oh, it's breaking the step

Would suggest returning an object: 
`const {verificationKey, isCached} = await step(`

Then adding a conditional after the step function to console log your message that it's cached.